### PR TITLE
Add device check before executing run-ios and run-android

### PR DESCRIPTION
**Pull request by enahum**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -57,10 +57,11 @@ run-ios: | start
 		@exit 1; \
 	fi
 
+run-ios: | check-device-ios start
 	@echo Running iOS app in development
 	@react-native run-ios --simulator="${SIMULATOR}"
```
Comment (hmhealey) [inline Makefile]: Shouldn't both ios and android skip the packager if it's ran by `make start`?

Comment (enahum) [reply Makefile]: the packager will not run, so this is fine

Comment (hmhealey) [reply Makefile]: Then why add --no-packager in `make run-android`?

Comment (enahum) [reply Makefile]: cause if the packager wasn't running before there seems to be a race condition and it ends up running twice

Comment (hmhealey) [reply Makefile]: Ah okay, so it's only a problem with the Android one. RN really does not work well with Android

**Pull request merged**